### PR TITLE
Fix to improve hostname handling

### PR DESCRIPTION
**Pull request by devpatha**
#### Why I did it
If config_db.json is missing hostname entry, hostname-config.sh ends up deleting existing entry too and hostname changes to default 'localhost'

#### How I did it
If hostname is not available from config_db.json, do not edit /etc/hostname and exit

#### How to verify it
with missing hostname entry in config_db.json, /etc/hostname entry is not cleared

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012 for Cisco 8102 
- [ ] 202106
- [ ] 202111
- [ ] 202205

### CODE REVIEW

File: files/image_config/hostname/hostname-config.sh [hunk 1]
```diff
@@ -3,6 +3,11 @@
 CURRENT_HOSTNAME=`hostname`
 HOSTNAME=`sonic-cfggen -d -v DEVICE_METADATA[\'localhost\'][\'hostname\']`
 
+if [ -z "$HOSTNAME" ] ; then
+       echo "Missing hostname in the config file!"
+       exit 0
+fi
```
Comment (yxieca) [inline files/image_config/hostname/hostname-config.sh:9]: Why not set a good default hostname instead?

Comment (devpatha) [reply files/image_config/hostname/hostname-config.sh:9]: shall we default to "sonic"?

**Review by yxieca — APPROVED** [review]

**Pull request merged**